singleflight/example_test.go

```diff
@@ -0,0 +1,37 @@
+package singleflight
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+func Example() {
+	// Spin up 5 goroutines, each of which calls a function that increments
+	// a counter. At the end, the counter will only increment once and the
+	// counter value (1) will be returned to all goroutines.
+	counter := 0
+	var g Group
+	var wg sync.WaitGroup
+	var results [5]int
+	c := make(chan bool)
+	for i := 0; i < 5; i++ {
+		wg.Add(1)
+		go func(i int) {
+			ctrval, _ := g.Do("key", func() (interface{}, error) {
+				counter++
```

Comment:
I don't like how this example basically teaches people how to write data races. You have to already understand how singleflight works to understand why this perhaps is safe.

Use atomic.AddInt32 instead? And then put a same-line `// atomic` comment on the `var counter int32 // atomic` line.